app/main.py

```diff
@@ -1,34 +1,99 @@
+from typing import Union
+
+
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(
+            self,
+            row: int,
+            column: int,
+            is_alive: bool = True
+    ) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(
+            self,
+            start: tuple[int],
+            end: tuple[int],
+            is_drowned: bool = False
+    ) -> None:
+        self.is_drowned = is_drowned
+        self.decks = []
+        self._create_ship(start, end)
+
+    def _create_ship(self, start: tuple[int], end: tuple[int]) -> None:
+        col_coord, row_coord = (start[0], end[0]), (start[1], end[1])
+        rows_range = range(min(col_coord), max(col_coord) + 1)
+        col_range = range(min(row_coord), max(row_coord) + 1)
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+        for row in rows_range:
+            for column in col_range:
+                self.decks.append(Deck(row, column))
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def get_deck(self, row: int, column: int) -> Union[Deck, None]:
```

Comment:
Change your return type annotation, instead of `Union` use `|`:
```suggestion
    def get_deck(self, row: int, column: int) -> Deck | None:
```
In case you face errors, use `from __future__ import annotations` import